scripts/build.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash -
+
+if [ -d node_modules ]
+    then
+        rm -rf node_modules
+fi
+
+if [ -d build ]
+    then
+        rm -rf build
+fi
+
+if [ -f grader.zip ]
+    then
+        rm grader.zip
+fi
+
+yarn install --production=true
+tsc
+cd build && zip -r --exclude=*terraform* ../grader.zip index.js graphics/ && cd ..
```

Comment:
Can't find `./graphics/*` in the zip file, may need a `-r` flag somewhere. Please verify the built zip file with `$ unzip -l grader.zip`.

Reply:
Strange... I checked the zip file on my machine and it contains `./graphics/*`. I'll have a look again.

Reply:
Here's the output of a freshly-built zip file with the node modules truncated:

```
Archive:  grader.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
     3623  2018-08-19 23:09   index.js
        0  2018-08-19 23:09   graphics/
   122764  2018-08-19 23:09   graphics/rune_library.js
   140554  2018-08-19 23:09   graphics/rune_library.js.map
        0  2018-08-19 23:09   node_modules/
        0  2018-08-19 23:09   node_modules/acorn/
    11096  2018-08-15 20:59   node_modules/acorn/CHANGELOG.md
        0  2018-08-19 23:09   node_modules/acorn/dist/
    47836  2018-08-15 20:59   node_modules/acorn/dist/acorn_loose.es.js
   179412  2018-08-15 20:59   node_modules/acorn/dist/acorn.js
    14926  2018-08-15 20:59   node_modules/acorn/dist/walk.js
    49813  2018-08-15 20:59   node_modules/acorn/dist/acorn_loose.js
        0  2018-08-15 20:59   node_modules/acorn/dist/.keep
...
        0  2018-08-19 23:09   node_modules/typescript/lib/tr/
   159296  2018-08-19 12:19   node_modules/typescript/lib/tr/diagnosticMessages.generated.json
     1103  2018-08-19 12:19   node_modules/typescript/lib/watchGuard.js
    10300  2018-08-19 12:19   node_modules/typescript/lib/lib.es2015.symbol.wellknown.d.ts
     9197  2018-08-19 12:19   node_modules/typescript/LICENSE.txt
     4965  2018-08-19 12:19   node_modules/typescript/README.md
---------                     -------
 41411914                     1525 files

```

The graphics folder is there together with its contents.

Reply:
```
ning:~/github/grader $ yarn build > /dev/null && git show --summary && unzip -l grader.zip | head
warning " > ts-jest@23.0.0" has unmet peer dependency "babel-core@^6.26.3 || ^7.0.0-0".              
warning " > ts-jest@23.0.0" has unmet peer dependency "babel-core@^6.26.3 || ^7.0.0-0".              
commit a3b66db41285e4acf590390a8b8495747d4cabb7 (HEAD -> support-runes, origin/support-runes)
Author: Thenaesh Elango <thenaeshelango@gmail.com>
Date:   Sun Aug 19 21:45:11 2018 +0800

    Ignore grader.zip

Archive:  grader.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
     3683  2018-08-20 00:05   index.js
        0  2018-08-20 00:05   node_modules/
        0  2018-08-20 00:05   node_modules/js-slang/
        0  2018-08-20 00:05   node_modules/js-slang/dist/
      868  2018-08-12 17:03   node_modules/js-slang/dist/interpreter.d.ts
     1635  2018-08-12 17:03   node_modules/js-slang/dist/types.js
    16768  2018-08-12 17:03   node_modules/js-slang/dist/interpreter.js.map
```

Alright, something weird is going on here

Reply:
Oh, so I think that you are zipping in a `./graphics` directory that was generated *previously* by tsc, probably when the target directory for `tsc` was still the project root. I never ran `tsc` when that was the case, so I don't have the `./graphics` directory (and it's not tracked by git either).

You should probably be zipping `./build/src/graphics` in there. Although, you will need to be careful about whether the `index.js` file can import the `graphics/*` libraries correctly (i.e. file path troubles). Would be best to unzip the `grader.zip` package somewhere and just run the `exports.runAll` function.